src/app/tests/TestCommandInteraction.cpp

```diff
@@ -588,6 +599,36 @@ void TestCommandInteraction::TestCommandSenderCommandSuccessResponseFlow(nlTestS
     NL_TEST_ASSERT(apSuite, ctx.GetExchangeManager().GetNumActiveExchanges() == 0);
 }
 
+void TestCommandInteraction::TestCommandSenderCommandAsyncSuccessResponseFlow(nlTestSuite * apSuite, void * apContext)
```

Comment:
Can you please add appropriate tests to `src/controller/tests/data_model/TestCommands.cpp` as well? That is a more representative 'end-to-end' test that contains actual representative server side logic, as well as async dispatch support for handling commands that correctly validates async message processing behavior as well.

Reply:
Done